eZ/Publish/Core/Repository/RoleService.php

```diff
@@ -434,6 +825,32 @@ public function loadRole($id)
     }
 
     /**
+     * Loads a role draft for the given identifier.
+     *
+     * @throws \eZ\Publish\API\Repository\Exceptions\InvalidArgumentException if the authenticated user is not allowed to read this role
+     * @throws \eZ\Publish\API\Repository\Exceptions\UnauthorizedException if the authenticated user is not allowed to read this role
+     * @throws \eZ\Publish\API\Repository\Exceptions\NotFoundException if a role with the given name was not found
+     *
+     * @param string $identifier
+     *
+     * @return \eZ\Publish\API\Repository\Values\User\RoleDraft
+     */
+    public function loadRoleDraftByIdentifier($identifier)
```

Comment:
> and for detecting role edit conflicts.

but where is this done now, is it done at several layers? cause here it is done on "Business Logic" [layer](https://doc.ez.no/pages/viewpage.action?pageId=11403666#eZPublish5Architecture-Introduction&Overview-Thetargetarchitecture:"eZPlatform"), aka the implementation of the API, so for that we only need to document exceptions to expect for this in create/update api, and have spi methods that allows us to check this (for drafts )